File: src/examples/addSignatory/overrideEvidenceProviderSettings.cs
using Criipto.Signatures;
using Criipto.Signatures.Models;

using (var client = new CriiptoSignaturesClient("{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"))
{
    // Setup document input
    var documents = new List<DocumentInput>{
        new DocumentInput {
            pdf = new PadesDocumentInput {
                title = "Dotnet Sample",
                blob = new Byte[64], // Should be the bytes of a PDF file
                storageMode = DocumentStorageMode.Temporary
            }
        }
    };

    // Setup signature order input
    var createSignatureOrderInput = new CreateSignatureOrderInput
    {
        title = "Dotnet Sample",
        documents = documents
    };

    // Create signature order
    var signatureOrder = await client.CreateSignatureOrder(createSignatureOrderInput);

    var criiptoVerifyEvidenceProviderId =
        signatureOrder.evidenceProviders.Find(ep => ep is CriiptoVerifySignatureEvidenceProvider)?.id ??
        throw new InvalidOperationException("Criipto Verify Evidence Provider not found");

    // Add signatory
    var signatory = await client.AddSignatory(
        signatureOrder.id,
        new AddSignatoryInput()
        {
            evidenceProviders = new List<SignatoryEvidenceProviderInput>
            {
                new() {
                    id = criiptoVerifyEvidenceProviderId,
                    criiptoVerify = new CriiptoVerifyProviderInput
                    {
                        acrValues = new List<string>
                        {
                            "urn:grn:authn:dk:mitid:substantial",
                            "urn:grn:authn:se:bankid:another-device:qr",
                            "urn:grn:authn:no:bankid:substantial"
                        },
                        message = "signatory message",
                        loginHint = "signatory login hint"
                    }
                }
            }
        }
    );
    Console.WriteLine(signatory.href); // Signing link, redirect user to this link, or send it in an email
}

File: src/examples/addSignatory/overrideEvidenceProviderSettings.py
from criipto_signatures import CriiptoSignaturesSDKSync
from criipto_signatures.models import (
  CreateSignatureOrderInput,
  DocumentInput,
  PadesDocumentInput,
  DocumentStorageMode,
  AddSignatoryInput,
  CriiptoVerifySignatureEvidenceProvider,
  CriiptoVerifyProviderInput,
  SignatoryEvidenceProviderInput,
)

client = CriiptoSignaturesSDKSync(
  "{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"
)

# Create signature order
signatureOrder = client.createSignatureOrder(
  CreateSignatureOrderInput(
    title="Python sample",
    documents=[
      DocumentInput(
        pdf=PadesDocumentInput(
          title="My document",
          blob=bytes("...", "utf-8"),  # bytes object, or a base64 encoded string
          storageMode=DocumentStorageMode.Temporary,
        )
      )
    ],
  )
)

criiptoVerifyEvidenceProviderId = next(
  (
    evidenceProvider.id
    for evidenceProvider in signatureOrder.evidenceProviders
    if isinstance(evidenceProvider, CriiptoVerifySignatureEvidenceProvider)
  ),
  None,
)

if criiptoVerifyEvidenceProviderId is None:
  raise Exception("Criipto Verify Evidence Provider not found")

# Add signatory
signatory = client.addSignatory(
  AddSignatoryInput(
    signatureOrderId=signatureOrder.id,
    evidenceProviders=[
      SignatoryEvidenceProviderInput(
        id=criiptoVerifyEvidenceProviderId,
        criiptoVerify=CriiptoVerifyProviderInput(
          acrValues=[
            "urn:grn:authn:dk:mitid:substantial",
            "urn:grn:authn:se:bankid:another-device:qr",
            "urn:grn:authn:no:bankid:substantial",
          ],
          message="signatory message",
          loginHint="signatory login hint",
        ),
      )
    ],
  )
)

print(
  signatory.href
)  # Signing link, redirect user to this link, or send it in an email
